cognite/client/_api/simulators/simulator_models.py

```diff
@@ -0,0 +1,162 @@
+
+
+class SimulatorModelsAPI(APIClient):
+    _RESOURCE_PATH = "/simulators/models"
+
+    def __init__(self, config: ClientConfig, api_version: str | None, cognite_client: CogniteClient) -> None:
+        super().__init__(config, api_version, cognite_client)
+        self._warning = FeaturePreviewWarning(api_maturity="beta", sdk_maturity="alpha", feature_name="Simulators")
+
+    def list(
+        self, limit: int = DEFAULT_LIMIT_READ, filter: SimulatorModelsFilter | dict[str, Any] | None = None
+    ) -> SimulatorModelList:
+        """`Filter Simulator Models <https://api-docs.cogheim.net/redoc/#tag/Simulator-Models/operation/filter_simulator_models_simulators_models_list_post>`_
+
+        List all simulation models
+
+        Args:
+            limit (int): The maximum number of simulator models to return. Defaults to 100.
+            filter (SimulatorModelsFilter | dict[str, Any] | None): The filter to narrow down simulator models.
+
+        Returns:
+            SimulatorModelList: List of simulator models
+
+        Examples:
+
+            List simulators:
+
+                    >>> from cognite.client import CogniteClient
+                    >>> client = CogniteClient()
+                    >>> res = client.simulators.list_models()
+
+        """
+        self._warning.warn()
+        return self._list(
+            method="POST",
+            limit=limit,
+            url_path="/simulators/models/list",
+            resource_cls=SimulatorModel,
+            list_cls=SimulatorModelList,
+            headers={"cdf-version": "beta"},
+            filter=filter.dump()
+            if isinstance(filter, SimulatorModelsFilter)
+            else filter
+            if isinstance(filter, dict)
+            else None,  # fix this
+        )
+
+    def retrieve(self, id: int | None = None, external_id: str | None = None) -> SimulatorModel | None:
+        """`Retrieve Simulator Model <https://api-docs.cogheim.net/redoc/#tag/Simulator-Models/operation/retrieve_simulator_model_simulators_models_byids_post>`_
+
+        Get a simulator model by id/externalId
+
+        Args:
+            id (int | None): The id of the simulator model.
+            external_id (str | None): The external id of the simulator model.
+
+        Returns:
+            SimulatorModel | None: Requested simulator model
+
+        Examples:
+
+            List simulators:
+
+                    >>> from cognite.client import CogniteClient
+                    >>> client = CogniteClient()
+                    >>> res = client.simulators.retrieve_model()
+
+        """
+        identifiers = IdentifierSequence.load(ids=id, external_ids=external_id).as_singleton()
+        return self._retrieve_multiple(
+            list_cls=SimulatorModelList,
+            resource_cls=SimulatorModel,
+            identifiers=identifiers,
+            resource_path="/simulators/models",
+            headers={"cdf-version": "beta"},
+        )
+
+    def list_revisions(
```

Comment:
Ideally the revisions should have it's own class and sit under models, to allow something like: `client.simulators.models.revisions.list()`

Reply:
This is the normal pattern adopted by the Python SDK at least.